src/main/java/org/broadinstitute/hellbender/tools/walkers/coverage/CoverageOutputWriter.java

```diff
@@ -0,0 +1,797 @@
+
+                // Create a cumulative coverage statistics output sink
+                DoCOutputType cumulativeStatisticsOut = new DoCOutputType(partition, DoCOutputType.Aggregation.cumulative, DoCOutputType.FileType.statistics);
+                outputs.put(cumulativeStatisticsOut, getOutputStream(outputBaseName, cumulativeStatisticsOut, separator));
+            }
+        }
+
+        if(!omitLocusTable) {
+            for(DoCOutputType.Partition partition: partitions) {
+                // Create a cumulative coverage counts output sink
+                DoCOutputType cumulativeCoverageCountsOut = new DoCOutputType(partition, DoCOutputType.Aggregation.cumulative, DoCOutputType.FileType.coverage_counts);
+                outputs.put(cumulativeCoverageCountsOut, getOutputStream(outputBaseName, cumulativeCoverageCountsOut, separator));
+
+                // Create a cumulative coverage portions output sink
+                DoCOutputType cumulativeCoverageProportionsOut = new DoCOutputType(partition, DoCOutputType.Aggregation.cumulative, DoCOutputType.FileType.coverage_proportions);
+                outputs.put(cumulativeCoverageProportionsOut, getOutputStream(outputBaseName, cumulativeCoverageProportionsOut, separator));
+            }
+        }
+    }
+
+    // Helper method to generate an output stream given a DoCOutputType Object and the base name
+    private static SimpleCSVWriterWrapperWithHeader getOutputStream(String outputBaseName, DoCOutputType depthSummaryByLocus, char separator) throws IOException {
+        return new SimpleCSVWriterWrapperWithHeader(new OutputStreamWriter(Files.newOutputStream((IOUtils.getPath(depthSummaryByLocus.getFilePath(outputBaseName))))), separator);
+    }
+
+    /**
+     * Initialize and write the output headers for the output streams that need to be continuously updated during traversal.
+     *
+     * @param sortedSamplesByPartition global map of partitionType to sorted list of samples associated with that partition
+     * @param coverageThresholds Threshold coverage level to use for reporting.
+     */
+    public void writeCoverageOutputHeaders(final Map<DoCOutputType.Partition, List<String>> sortedSamplesByPartition, final List<Integer> coverageThresholds) {
+        this.coverageThresholds = coverageThresholds;
+
+        if ( ! omitDepthOutput ) { // print header
+            writePerLocusDepthOutputSummaryHeader(sortedSamplesByPartition);
+        }
+
+        // write
+        if ( includeGeneOutput && partitions.contains(DoCOutputType.Partition.sample) ) {
+            final SimpleCSVWriterWrapperWithHeader geneSummaryOut = getCorrectOutputWriter(DoCOutputType.Partition.sample, DoCOutputType.Aggregation.gene, DoCOutputType.FileType.summary);
+            geneSummaryOut.addHeaderLine(getIntervalSummaryHeader("Gene", sortedSamplesByPartition.get(DoCOutputType.Partition.sample)));
```

Comment:
I will punt on this as there are a LOT of string literals in this class and I think it will actually confuse things more than it helps if taken to its logical extreme